File: src/week2/report2/Reservation.java
package week2.report2;

public class Reservation {
  static final Reservation NONE = new Reservation(null, null, null, null, 0);
  final Theater theater;
  final Movie movie;
  final Auditorium auditorium;
  final Screening screening;
  final int count;

  Reservation(Theater theater, Movie movie, Auditorium auditorium , Screening screening, int audienceCount) {
    this.theater = theater;
    this.movie = movie;
    this.auditorium = auditorium;
    this.screening = screening;
    this.count = audienceCount;
  }
}

File: Swift/week2/Week2/Week2/Example/Reservation.swift
//
//  Reservation.swift
//  Week2
//
//  Created by tskim on 18/10/2019.
//  Copyright © 2019 assin. All rights reserved.
//

import Foundation

class Reservation {
    
    let theater: Theater?
    let  movie: Movie?
    let screening: Screening?
    let count: Int

    public init(theater: Theater?, movie: Movie?, screening: Screening?, count: Int) {
        self.theater = theater
        self.movie = movie
        self.screening = screening
        self.count = count
    }
}

extension Reservation {
    static let None = Reservation(theater: nil, movie: nil, screening: nil, count: 0)
}
